lib/client.js

```diff
@@ -2807,6 +2807,82 @@ MatrixClient.prototype.register = function(username, password,
  */
 MatrixClient.prototype.requestRegisterEmailToken = function(email, clientSecret,
                                                     sendAttempt, nextLink, callback) {
+    return this._requestTokenFromEndpoint(
+        "/register/email/requestToken",
+        email, clientSecret, sendAttempt, nextLink, callback
+    );
+};
+
+/**
+ * Requests an email verification token for the purposes of adding a
+ * third party identifier to an account.
+ * This API proxies the Identity Server /validate/email/requestToken API,
+ * adding specific behaviour for the addition of email addresses to an
+ * account. Specifically, if an account with
+ * the given email address already exists, it will either send an email
+ * to the address informing them of this or return M_THREEPID_IN_USE
+ * (which one is up to the Home Server).
+ *
+ * requestEmailToken calls the equivalent API directly on the ID server,
+ * therefore bypassing the email addition specific logic.
+ *
+ * @param {string} email As requestEmailToken
+ * @param {string} clientSecret As requestEmailToken
+ * @param {number} sendAttempt As requestEmailToken
+ * @param {string} nextLink As requestEmailToken
+ * @param {module:client.callback} callback Optional. As requestEmailToken
+ * @return {module:client.Promise} Resolves: As requestEmailToken
+ */
+MatrixClient.prototype.request3pidAddEmailToken = function(email, clientSecret,
```

Comment:
Descriptive function name but it feels a bit mouthy. \*thinks a bit\* - It's the "3pid add" bit I don't like. Maybe flip it around: "add 3pid"? E.g:
```
requestAdd3pidEmailToken
```

That matches the description better:

> Requests an email verification token for the purposes of **adding a third party identifier** to an account.